Data/Row/Variants.hs

```diff
@@ -306,3 +314,112 @@ fromLabels mk = getCompose $ metamorph' @_ @ρ @c @(Const ()) @(Compose f Var) @
         doCons l (Left _) = Compose $ unsafeMakeVar l <$> mk l --This case should be impossible
         doCons l (Right (Compose v)) = Compose $
           unsafeMakeVar l <$> mk l <|> unsafeInjectFront <$> v
+
+{--------------------------------------------------------------------
+  Native data type compatibility
+--------------------------------------------------------------------}
+
+-- $native
+-- The 'toNative' and 'fromNative' functions allow one to convert between
+-- 'Var's and regular Haskell data types ("native" types) that have the same
+-- number of constructors such that each constructor has one field and the same
+-- name as one of the options of the 'Var', which has the same type as that field.
+-- That said, they do not compose to form the identity because 'fromNative' allows
+-- constructors to be added: a variant with excess options can still be transformed
+-- to a native type, but when the native type is converted to a variant, the
+-- options are exactly transformed.  The only requirement is that
+-- the native Haskell data type be an instance of 'Generic'.
+--
+-- For example, consider the following simple data type:
+--
+-- >>> data Pet = Dog {age :: Int} | Cat {age :: Int} deriving (Generic, Show)
+--
+-- Then, we have the following:
+--
+-- >>> toNative $ IsJust (Label @"Dog") 3 :: Pet
+-- Dog {age = 3}
+-- >>> V.fromNative $ Dog 3 :: Var ("Dog" .== Int .+ "Cat" .== Int)
+-- {Dog=3}
+--
+-- The 'fromNativeExact' function is a more restricted version of 'fromNative' that
+-- does not allow options to be added; in other words, the options in the variant
+-- must exactly match the constructors in the data type.  Because of this,
+-- 'fromNativeExact' and 'toNative' compose to form the identity function.
+
+
+-- | Conversion helper to bring a variant back into a Haskell type. Note that the
+-- native Haskell type must be an instance of 'Generic'.
+class ToNative a ρ where
+  toNative' :: Var ρ -> a x
+
+instance ToNative cs ρ => ToNative (G.D1 m cs) ρ where
+  toNative' = G.M1 . toNative'
+
+instance ToNative G.V1 Empty where
+  toNative' = impossible
+
+instance (KnownSymbol name, ρ ≈ name .== t)
+    => ToNative (G.C1 ('G.MetaCons name fixity sels)
+                (G.S1 ('G.MetaSel n p s l) (G.Rec0 t))) ρ where
```

Comment:
Can the `'G.MetaSel n p s l` rather be merely `m`? I think it would help a reader following both this and `Data.Row.Records.ToNative` — here we care about `MetaCons`; there we care about `MetaSel`.

Reply:
Yes!  Good idea for clarity -- I'm putting that in.